addons/website_sale_stock/models/product_ribbon.py

```diff
@@ -18,15 +18,10 @@ class ProductRibbon(models.Model):
         ),
     )
 
-    def _get_applicable_ribbon(self, product, product_prices):
-        """
-        Override of `website_sale` to check if the product is eligible for out of stock ribbon.
-        """
-        for ribbon in self:
-            if (
-                ribbon.assign == 'out_of_stock'
-                and not product.product_tmpl_id.allow_out_of_stock_order
-                and product._is_sold_out()
-            ):
-                return ribbon
-        return super()._get_applicable_ribbon(product, product_prices)
```

Comment:
@dere-odoo This was wrong, as it didn't take the sequence into account. If there were an `out_of_stock` ribbon in the list, it would always have been applied first, no matter the sequence defined on the ribbon.